block/manager.go

```diff
@@ -170,36 +183,78 @@ func (m *Manager) GetFraudProofOutChan() chan *abci.FraudProof {
 }
 
 // AggregationLoop is responsible for aggregating transactions into rollup-blocks.
-func (m *Manager) AggregationLoop(ctx context.Context) {
-	initialHeight := uint64(m.genesis.InitialHeight)
-	height := m.store.Height()
-	var delay time.Duration
+func (m *Manager) AggregationLoop(ctx context.Context, ingress chan []byte, txVal func([]byte) bool, progressive bool) {
+	if !progressive {
+		initialHeight := uint64(m.genesis.InitialHeight)
+		height := m.store.Height()
+		var delay time.Duration
+
+		// TODO(tzdybal): double-check when https://github.com/celestiaorg/rollmint/issues/699 is resolved
+		if height < initialHeight {
+			delay = time.Until(m.genesis.GenesisTime)
+		} else {
+			delay = time.Until(m.lastState.LastBlockTime.Add(m.conf.BlockTime))
+		}
 
-	// TODO(tzdybal): double-check when https://github.com/celestiaorg/rollmint/issues/699 is resolved
-	if height < initialHeight {
-		delay = time.Until(m.genesis.GenesisTime)
-	} else {
-		delay = time.Until(m.lastState.LastBlockTime.Add(m.conf.BlockTime))
-	}
+		if delay > 0 {
+			m.logger.Info("Waiting to produce block", "delay", delay)
+			time.Sleep(delay)
+		}
 
-	if delay > 0 {
-		m.logger.Info("Waiting to produce block", "delay", delay)
-		time.Sleep(delay)
-	}
+		timer := time.NewTimer(0)
 
-	timer := time.NewTimer(0)
+		for {
+			select {
+			case msg := <-ingress:
```

Comment:
non-progressive sequencer should not expect tx in ingress. remove this. it should proceed to building blocks at regular interval, even if no txns.